### DISCUSSION TITLE
Initializing a sensor enable gpio before usage

### DISCUSSION THREAD

**Discussion by vulporuza** [zephyrproject-rtos/zephyr#34174] [Q&A] [answered]
Hi, 

First of all I am not yet confortable with all device tree concepts but very interested.

My problem is to use a sensor, addressed through I2C, but mapped on board to be enable using a GPIO (nRF52 based board).
In my device tree, I declare it as following and all seems ok:
```
	lsm303agr-accel@19 {
		compatible = "st,lis2dh", "st,lsm303agr-accel";
		status = "okay";
		reg = <0x19>;
		label = "LSM303AGR-ACCEL";
		irq-gpios = <&gpio1 10 GPIO_ACTIVE_HIGH>;
	};
```

My board defconfig also mentions sensors and lis2dh requirement, all is built.
But on boot I have an error on the I2C probing of my sensor, due to the enable GPIO unset.

2 questions:
- Is there a way to declare an "enable pin" in the dts that I missed ? (I have read all dts bindings but I can't find it, is there an equivalent to csn pin over SPI ?)
- Is there a way to run a POST_KERNEL routine allowing to configure the enable gpio as output high between GPIO init (priority CONFIG_KERNEL_INIT_PRIORITY_DEFAULT) and sensor init (priority CONFIG_SENSOR_INIT_PRIORITY).

I am sure that zephyr has a feature for this kind of problem, but I do not find solution.

Thanks in advance for any help :)

Robin-Charles

**Comment by henrikbrixandersen** [comment, marked as answer]
Depending on the actual "enable" pin design (is it a power regulator? is it a pin present on the sensor for enabling? something else?) there are a couple of options:
- Use a regulator for modelling the enable pin: https://docs.zephyrproject.org/latest/reference/peripherals/regulators.html
- Add a `enable-gpios` property to the devicetree binding for the given sensor and modify the driver to support asserting this pin when the device needs to be enabled
- Add a `board.c` file for asserting the pin at board boot time (see e.g. https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/arty/board.c)
